client-src/default/utils/createSocketUrl.js

```diff
@@ -3,41 +3,60 @@
 /* global self */
 
 const url = require('url');
-const querystring = require('querystring');
 const getCurrentScriptSource = require('./getCurrentScriptSource');
 
-function createSocketUrl(resourceQuery) {
+function createSocketUrl(resourceQuery, currentLocation) {
   let urlParts;
 
   if (typeof resourceQuery === 'string' && resourceQuery !== '') {
     // If this bundle is inlined, use the resource query to get the correct url.
-    // strip leading `?` from query string
-    urlParts = url.parse(resourceQuery.substr(1));
+    // format is like `?http://0.0.0.0:8096&sockPort=8097&sockHost=localhost`
+    urlParts = url.parse(
+      resourceQuery
+        // strip leading `?` from query string to get a valid URL
+        .substr(1)
+        // replace first `&` with `?` to have a valid query string
+        .replace('&', '?'),
+      true
+    );
   } else {
     // Else, get the url from the <script> this file was called with.
     const scriptHost = getCurrentScriptSource();
-    urlParts = url.parse(scriptHost || '/', false, true);
+    urlParts = url.parse(scriptHost || '/', true, true);
```

Comment:
Why we need parse query string here?

Reply:
This is related to the comment above. Having a full URL including parsed query string saves from another `querystring.parse(urlParts.path)` later on.